src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1536,14 +1536,18 @@ end
 
 """
     hcat(df::AbstractDataFrame...;
-         makeunique::Bool=false, copycols::Bool=true)
+         makeunique::Bool=false, mergeduplicates=nothing, copycols::Bool=true)
 
 Horizontally concatenate data frames.
 
 If `makeunique=false` (the default) column names of passed objects must be unique.
 If `makeunique=true` then duplicate column names will be suffixed
 with `_i` (`i` starting at 1 for the first duplicate).
 
+If `makeunique=false` and `mergeduplicates` is a Function then duplicate column names 
+will be combined by this function with the column named overwritten by the results of 
+the function on all values from the duplicated column(s).
```

Comment:
It is not clear what this sentence means. Probably a native speaker would be better to say how to fix it.

What I can say is how this will work `mergeduplicates(mergeduplicates(x,y),z)`

Reply:
Can I say "ouch". I am very much a native English speaker. That said I've struggled with this wording to make it clearer...  more struggle then.

Reply:
Would the following be clearer?

> If `makeunique=false` and `mergeduplicates` is a `Function` then duplicate columns 
will be combined by invoking the function with all values from those columns.
e.g. `mergeduplicates=coalesce` will use the first non-missing value.

Reply:
The crucial issue is, as I have commented, that the behavior or `mergeduplicates` will differ depending on the function in which it is used (unless we change the current implementation).

In functions like `insertcols` or `DataFrame` it will pass all duplicate columns as consecutive arguments to a SINGLE CALL to `mergeduplicates` function.

In functions like `hcat` or `*join` it will perform a recursive call taking at most two duplicate columns at a time.